tests/test_subset_contigs_variants.py

```diff
@@ -0,0 +1,522 @@
+    1. Same set of contig indices found
+    2. Same orientation for each contig
+    """
+    naive_keys = set(naive_result.keys())
+    other_keys = set(other_result.keys())
+    
+    all_passed = True
+
+    # Check 1: Same contig indices
+    if naive_keys != other_keys:
+        missing = naive_keys - other_keys
+        extra = other_keys - naive_keys
+        print(f"  ✗ {name}: Different contig indices!")
+        if missing:
+            print(f"    Missing {len(missing)} matches: {list(missing)[:5]}...")
+        if extra:
+            print(f"    Extra {len(extra)} matches: {list(extra)[:5]}...")
+        all_passed = False
+    
+    # Check 2: Same orientations for matching indices
+    common_keys = naive_keys & other_keys
+    orientation_mismatches = []
+    for key in common_keys:
+        if naive_result[key] != other_result[key]:
+            orientation_mismatches.append(key)
+    
+    if orientation_mismatches:
+        print(f"  ✗ {name}: Orientation mismatches for {len(orientation_mismatches)} contigs!")
+        for key in orientation_mismatches[:5]:
+            print(f"    Contig {key}: naive={naive_result[key].value}, {name}={other_result[key].value}")
+        if len(orientation_mismatches) > 5:
+            print(f"    ... and {len(orientation_mismatches) - 5} more")
+        all_passed = False
+    
+    if all_passed:
+        print(f"  ✓ {name}: All {len(naive_result)} matches verified (indices + orientations)")
+    
+    return all_passed
+
+
+def run_correctness_tests(include_bowtie2: bool = True) -> bool:
+    """
+    Run thorough correctness tests before benchmarking.
+    
+    Tests:
+    1. Empty inputs
+    2. No matches
+    3. All matches
+    4. Forward orientation matches
+    5. Reverse complement matches
+    6. Mixed orientations
+    7. Multiple seeds matching same contig
+    8. Palindromic seeds
+    
+    Returns True if all tests pass.
+    """
+    print("=" * 80)
+    print("CORRECTNESS TESTS")
+    print("=" * 80)
+    print()
+    
+    all_passed = True
+    
+    # Test 1: Empty inputs
+    print("Test 1: Empty inputs...")
+    empty_seeds: List[Seq] = []
+    empty_contigs: List[SeqRecord] = []
+    
+    assert contig_ids_by_seed(empty_contigs, empty_seeds) == {}
+    assert _contig_ids_by_seed_ahocorasick(empty_contigs, empty_seeds) == {}
+    print("  ✓ Empty inputs handled correctly")
+    
+    # Test 2: Known forward match
+    print("\nTest 2: Known forward match...")
+    seed = Seq("GATTACA")
+    contig = SeqRecord(Seq("AAAAAAGATTACATTTTTT"), id="contig_0")
+    
+    naive_result = contig_ids_by_seed([contig], [seed])
+    ac_result = _contig_ids_by_seed_ahocorasick([contig], [seed])
+    
+    assert 0 in naive_result, "Naive should find match"
+    assert naive_result[0] == SeqOrientation.FORWARD, "Should be forward orientation"
+    assert 0 in ac_result, "Aho-Corasick should find match"
+    assert ac_result[0] == SeqOrientation.FORWARD, "Should be forward orientation"
```

Comment:
Rather than having two sets of asserts in each test, we could have one correctness test and parameterize by the method called?